models/agent_helpers.go

```diff
@@ -226,21 +214,7 @@ func FindPMMAgentsForService(q *reform.Querier, serviceID string) ([]*Agent, err
 // FindAgentsByServiceIDAndAgentType find agents by service_id and agent_type.
 //nolint:unused
 func FindAgentsByServiceIDAndAgentType(q *reform.Querier, serviceID string, agentType AgentType) ([]*Agent, error) {
-	asMap, err := q.SelectAllFrom(AgentServiceView, "WHERE service_id = $1", serviceID)
-	if err != nil {
-		return nil, status.Errorf(codes.FailedPrecondition, "Failed to select AgentService map, reason: %v", err)
-	}
-
-	aIds := make([]interface{}, len(asMap))
-	for _, str := range asMap {
-		row := str.(*AgentService)
-		aIds = append(aIds, row.AgentID)
-	}
-
-	// Last, find all pmm-agents.
-	ph := strings.Join(q.Placeholders(1, len(aIds)), ", ")
-	atail := fmt.Sprintf("WHERE agent_id IN (%s) AND agent_type = '%s'", ph, agentType) //nolint:gosec
-	structs, err := q.SelectAllFrom(AgentTable, atail, aIds...)
+	structs, err := q.SelectAllFrom(AgentTable, "WHERE service_id = $1 AND agent_type = $2", serviceID, agentType)
```

Comment:
I think using FindAgentsForService there and filter by type will give us a better error message. Let's add a unit test for it.

Reply:
Since this method is used only in one place and we already have filter by pmm-agent there I decided to remove this method at all and move logic there.